unixfs/io/pbdagreader.go

```diff
@@ -264,6 +283,11 @@ func (dr *PBDagReader) Seek(offset int64, whence int) (int64, error) {
 			return offset, nil
 		}
 
+		// error out if we have links but not blocksizes
+		if len(pb.Blocksizes) == 0 && len(dr.node.Links()) > 0 {
```

Comment:
so this means that seeking through a unixfs dag with no blocksizes set will error out. We *could* dumb-seek through by just reading a bunch, but that might produce unexpected perf degredations.

@mib-kd743naq does this work for you (erroring out on seek of no-blocksizes)?

Reply:
@whyrusleeping I am not sure how to answer this, as I do not have a good mental model *where* does the seeker get invoked.

Coming from the other side: what I need is to keep the ability of [the test cases for #4286](https://ipfs.io/ipfs/QmZL4wivfYBEUutxksKgbpPfFkj1tGNMESYLFF2MWcST8Y/) to be:
- fully traversed/displayed [via the web gateway](https://ipfs.io/ipfs/QmZL4wivfYBEUutxksKgbpPfFkj1tGNMESYLFF2MWcST8Y)
- can be downloaded through `ipfs get QmZL4wivfYBEUutxksKgbpPfFkj1tGNMESYLFF2MWcST8Y`

If both of these continue to work - I am happy ;)

@kevina if you need further size-less testcase construction - just let me know roughly what tree shape you are interested in

Reply:
@mib-kd743naq The tests cases added in #4286 now pass and I manually tested that the hash QmZL4wivfYBEUutxksKgbpPfFkj1tGNMESYLFF2MWcST8Y works with both the gateway and `get`.

Reply:
Where this will fail is if you use the `Range:` header to retrieve partial content.  If you do something like `GET /ipfs/QmTTDNxNJUqF5PZicJqjDjsUed1bHaPWwFdEMvb1iY93Ur HTTP/1.1` with `Range:bytes=3-` you get back `HTTP/1.1 416 Requested Range Not Satisfiable`.

This can be fixed by doing a dummy seek, or alternatively not have the gateway not return a `Content-Range` response in the header.